contracts/MultiPartyEscrow.sol

```diff
@@ -0,0 +1,272 @@
+pragma solidity ^0.4.24;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+
+contract MultiPartyEscrow {
+    
+    //it seems we don't need SafeMath
+   //using SafeMath for uint256;
+    
+
+    //TODO: we could use uint64 for replica_id and nonce (it could be cheaper to store but more expensive to operate with)
+
+    //the full ID of "atomic" payment channel = "[this, channel_id, nonce]"
+    struct PaymentChannel {
+        address sender;      // The account sending payments.
+        address recipient;    // The account receiving the payments.
+        uint256 replica_id;  // id of particular service replica
```

Comment:
Code Style: I think we should adhere to the existent codebase (camelCase) and in general keep code style changes as separated PRs if needed.

Reply:
We should define the style... (I will follow it, even though I don't like camelCase :) ). I will open an Issue.

Reply:
Yeah, I mean it's not about personal preference.  If we decide to refactor the style, let's do it in a separate merge request.

As a "golden" rule, we should follow closely the master practices/styles which already exists, to make it consistent.

Reply:
@tiero, do you mean some common Solidity code style practices? If so could you please share link to it at https://github.com/singnet/platform-contracts/issues/43.

Reply:
Nope, I mean the code already merged in master is the style to follow. Always.

It's better to change the majority of old code to adapt to this PR or make sure this looks consistent with the already merged code? ;)

If @pennachin wants to change the code style and he allows to spend time on that ok to me to work on #43

Reply:
Please define codestyle in https://github.com/singnet/platform-contracts/issues/43 .  I cannot fix codestyle without definition of codestyle... What exactly? Only names of public function or what?  All those discussion is starting really pulling me backward..

Reply:
Wait, there are two separate tasks here:

1) Merge this pull request
2) Define code-style

The 1 cannot go through here, because we want a consistent code style. This PR has a different style introduced in naming variables/functions like this (name_of_var) instead of the existent nameOfVar camelCase style. 

The 2 came out since (maybe) we want to change and write in stone the rules (in order to setup a solidity linter/formatter in CI I presume)

The 1 and 2 can go in parallel, for the 1 the most obvious and quicker solution to me would adhere to the existent codebase, and not the other way. Moreover, I do think it's better to isolate code style changes in a specific PR.

Reply:
@tiero , am I right that converting name of functions and variables to camelCase will be enough for this PR?

Reply:
Exactly.

Reply:
And camelCase is the "official" style advocated by the Solidity docs.